addons/block_code/ui/picker/categories/category_factory.gd

```diff
@@ -336,16 +334,31 @@ static func get_built_in_categories(_class_name: String) -> Array[BlockCategory]
 			props = ["modulate"]
 
 		"RigidBody2D":
-			# On body entered
+			var b = BLOCKS["entry_block"].instantiate()
+			b.block_format = "On body entered [body: NODE_PATH]"
+			# HACK: Blocks refer to nodes by path but the callback receives the node itself;
+			# convert to path
+			b.statement = "func _on_body_entered(_body: Node):\n\tvar body: NodePath = _body.get_path()"
+			b.signal_name = "body_entered"
+			block_list.append(b)
```

Comment:
I actually don't think this will work for RigidBody2D. The [documentation for the `body_entered` signal says](https://docs.godotengine.org/en/stable/classes/class_rigidbody2d.html#class-rigidbody2d-signal-body-entered):

> Requires [contact_monitor](https://docs.godotengine.org/en/stable/classes/class_rigidbody2d.html#class-rigidbody2d-property-contact-monitor) to be set to true and [max_contacts_reported](https://docs.godotengine.org/en/stable/classes/class_rigidbody2d.html#class-rigidbody2d-property-max-contacts-reported) to be set high enough to detect all the collisions.

But those properties default to false and 0 respectively.

Reply:
(Of course the game developer can set those properties but it's not obvious from the block that you have to do so.)

Reply:
Yes, at least we need to make sure the contact monitor is enabled in the demo Pong project.